packages/filesystem/src/browser/file-tree/file-tree-widget.tsx

```diff
@@ -72,16 +73,25 @@ export class FileTreeWidget extends TreeWidget {
     }
 
     protected createNodeAttributes(node: TreeNode, props: NodeProps): React.Attributes & React.HTMLAttributes<HTMLElement> {
-        const elementAttrs = super.createNodeAttributes(node, props);
-        return {
-            ...elementAttrs,
+        const elementAttrs = {
+            ...super.createNodeAttributes(node, props),
             draggable: FileStatNode.is(node),
             onDragStart: event => this.handleDragStartEvent(node, event),
             onDragEnter: event => this.handleDragEnterEvent(node, event),
             onDragOver: event => this.handleDragOverEvent(node, event),
             onDragLeave: event => this.handleDragLeaveEvent(node, event),
             onDrop: event => this.handleDropEvent(node, event)
-        };
+        } as React.Attributes & React.HTMLAttributes<HTMLElement>;
+        const title = this.getNodeTooltip(node);
+        if (title) {
+            elementAttrs.title = title;
+        }
+        return elementAttrs;
+    }
+
+    protected getNodeTooltip(node: TreeNode): string | undefined {
```

Comment:
```ts
   protected getNodeTooltip(node: TreeNode): string | undefined {
        const uri = UriSelection.getUri(node)
        return uri && uri.path.toString();
    }
```
- a file tree node already has a URI
- when you work with URIs, everything encoded/decoded automatically, no need to use any other apis